kong/api/endpoints.lua

```diff
@@ -304,22 +304,28 @@ end
 local function get_collection_endpoint(schema, foreign_schema, foreign_field_name, method)
   return not foreign_schema and function(self, db, helpers)
     local next_page_tags = ""
+    local next_page_size = ""
 
     local args = self.args.uri
     if args.tags then
       next_page_tags = "&tags=" .. escape_uri(type(args.tags) == "table" and args.tags[1] or args.tags)
     end
 
+    if type(args.size) == "number" then
+      next_page_size = "&size=" .. args.size
+    end
+
     local data, _, err_t, offset = page_collection(self, db, schema, method)
     if err_t then
       return handle_error(err_t)
     end
 
-    local next_page = offset and fmt("/%s?offset=%s%s",
+    local next_page = offset and fmt("/%s?offset=%s%s%s",
                                      schema.admin_api_name or
                                      schema.name,
                                      escape_uri(offset),
-                                     next_page_tags) or null
+                                     next_page_tags,
+                                     next_page_size) or null
```

Comment:
Can you please include a test for this?

Reply:
Of course! But as I commented earlier, I didn't find any other automated tests for the next_page property. As the tests are segregated, I'm not sure how to proceed and in which file to put this test. If you direct me, I do it quickly :smile:

Reply:
Please take a look at the tests in https://github.com/Kong/kong/tree/master/spec/02-integration/04-admin_api and add in a test that asserts that the response has the appropriate field set to the value that you expect.

Reply:
I hope I did it the right way and wanted :smile: